lib/on-contract-event.js

```diff
@@ -0,0 +1,100 @@
+// eth_getFilterChanges isn't fully supported by lotus, see
+// https://filecoinproject.slack.com/archives/CRK2LKYHW/p1705404161018769
+// Therefore we need to poll for events
+
+import { ethers } from 'ethers'
+import timers from 'node:timers/promises'
+
+const MIN_POLL_INTERVAL = 10_000
```

Comment:
Why would a shorter polling interval be beneficial? Ie how could it be not fast enough?

Reply:
With 10-second intervals, we will be notified about new events up to 10 seconds after the network emits them. I guess that's fine for spark-evaluate.

IIRC, Filecoin has 30 seconds block time, so we will process the events before the next block starts 👍🏻 

For spark-api (and checker nodes running in Stations), this delay increases the number of jobs that may fall into an incorrect round because the client "thought" we were still in round N-1, but the network has already moved to round N. That's not a concern here in spark-evaluate, so all is good.

Reply:
Cool, thank you for validating my assumptions 🙏